server/id/reusable_id_test.go

```diff
@@ -0,0 +1,42 @@
+package id
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAlloc(t *testing.T) {
+	re := require.New(t)
+	cxt := context.Background()
+	// IDs: [1,2,3,5,6]
+	alloc := NewReusableAllocatorImpl([]uint64{1, 2, 3, 5, 6}, uint64(1))
```

Comment:
We need more cases for this allocator. Try to add some corner cases, e.g. existIDs is empty.